llama-index-integrations/vector_stores/llama-index-vector-stores-lancedb/llama_index/vector_stores/lancedb/base.py

```diff
@@ -343,13 +342,13 @@ def add(
             data.append(append_data)
             ids.append(node.node_id)
 
-        if self._table is None:
-            self._table = self._connection.create_table(
-                self._table_name, data, mode=self.mode
-            )
-        else:
+            if self._table is None:
+                self._table = self._connection.create_table(
```

Comment:
wait, why did we tab this back into the for-loop? Won't this miss a ton of data? (This would presumably only by run once on the first iteration of the loop, rather than at the end once `data` is fully built)

Reply:
thats a typo , thanks for pointing that out, I am making the resolution.